[improve][doc] Clarify maven running environments under windows (#18373)
apache/pulsar: README.md

- Maven 3.6.1+
- zip

> > Notes:

Reviewer: ```suggestion
> **Note**:
```

Trying to make use of GitHub rendering like:

<img width="581" alt="image" src="https://user-images.githubusercontent.com/18818196/200887060-c93197b8-38be-4a74-ba0b-4ff355c78f28.png">

> > * This project includes a [Maven Wrapper](https://maven.apache.org/wrapper/) that can be used instead of a system installed Maven.
> > Use it by replacing `mvn` by `./mvnw` on Linux and `mvnw.cmd` on Windows in the commands below.    
> > * It's better to use CMD rather than Powershell on Windows, since maven will activate windows profile and run `rename-netty-native-libs.cmd`.

Reviewer: ```suggestion
>
> This project includes a [Maven Wrapper](https://maven.apache.org/wrapper/) that can be used instead of a system-installed Maven.
> Use it by replacing `mvn` by `./mvnw` on Linux and `mvnw.cmd` on Windows in the commands below.    
>
> It's better to use CMD rather than Powershell on Windows. Because maven will activate the `windows` profile which runs `rename-netty-native-libs.cmd`.
```

Improve statement and format.
